packages/twitch-pubsub-client/src/Messages/PubSubSubscriptionMessage.ts

```diff
@@ -32,26 +34,90 @@ export default class PubSubSubscriptionMessage {
 		this._twitchClient = twitchClient;
 	}
 
-	// these following three methods can't use isGift because it messes up type inference
 	/**
 	 * The ID of the user subscribing to the channel.
 	 */
 	get userId() {
-		return this._data.context === 'subgift' ? this._data.recipient_id : this._data.user_id;
+		return this.isGift || this.isAnonymous
```

Comment:
Please don't use the other getters internally - then you don't have to use type assertions either.